### PR TITLE
Dual Stack support for vSphere

### LINKED ISSUE #2491: Add dual-stack IPv6 support for vSphere

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds dual stack support for vSphere.

**Which issue(s) this PR fixes**:

Fixes #2491

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:
For E2E tests: https://github.com/kubermatic/kubeone/issues/2821

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add dual stack support for vSphere
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: examples/terraform/vsphere/variables.tf [hunk 1]
```diff
@@ -249,3 +249,17 @@ variable "is_vsphere_enterprise_plus_license" {
...
+  # valid choices are:
+  # * IPv4
+  # * IPv4+IPv6 (dual-stack)
+  default = "IPv4"
+  type    = string
+  validation {
+    condition     = can(regex("^(IPv4|IPv4\\+IPv6)$", var.ip_family))
+    error_message = "Unsupported IP Family specified."
+  }
+}
```
Comment (xmudrii) [inline examples/terraform/vsphere/variables.tf:265]: Nit: please add a new line.

File: pkg/apis/kubeone/helpers.go [hunk 2]
```diff
@@ -493,6 +493,10 @@ func (c ClusterNetworkConfig) HasIPv6() bool {
 	return c.IPFamily == IPFamilyIPv6 || c.IPFamily == IPFamilyIPv4IPv6 || c.IPFamily == IPFamilyIPv6IPv4
 }
 
+func (c ClusterNetworkConfig) IsDualStack() bool {
+	return c.IPFamily == IPFamilyIPv4IPv6 || c.IPFamily == IPFamilyIPv6IPv4
+}
```
Comment (xmudrii) [inline pkg/apis/kubeone/helpers.go:498]: Why do we need this function when we already have `IsDualstack` function below?

File: pkg/scripts/funcs.go [hunk 3]
```diff
@@ -60,6 +60,13 @@ var (
 		net.bridge.bridge-nf-call-ip6tables = 1
 		net.bridge.bridge-nf-call-iptables  = 1
 		net.ipv4.ip_forward                 = 1
+		{{- if .IPV6_ENABLED }}
+		net.ipv6.conf.all.forwarding 		= 1
+		# Configure Linux to accept router advertisements to ensure the default
+		# IPv6 route is not removed from the routing table when the Docker service starts.
+		# For more information: https://github.com/docker/for-linux/issues/844
+		net.ipv6.conf.all.accept_ra		= 2
```
Comment (xmudrii) [inline pkg/scripts/funcs.go:68]: Is this relevant to us given that we don't support Docker?

Comment (ahmedwaleedmalik) [reply pkg/scripts/funcs.go:68]: Yeah it's still relevant.

File: addons/ccm-vsphere/ccm-vsphere.yaml [hunk 4]
```diff
@@ -188,8 +188,10 @@ spec:
           args:
             - --cloud-provider=vsphere
             - --cloud-config=/etc/cloud/cloud-config
-{{ if .Config.CABundle }}
           env:
+            - name: ENABLE_ALPHA_DUAL_STACK
+              value: "{{ .Config.ClusterNetwork.IPFamily.IsDualStack }}"
```
Comment (xmudrii) [inline addons/ccm-vsphere/ccm-vsphere.yaml:193]: I think the function is called `IsDualstack` (lowercase `s`).

Comment (ahmedwaleedmalik) [reply addons/ccm-vsphere/ccm-vsphere.yaml]: Updated

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve
